### PR TITLE
QueueBase improvements

### PR DESCRIPTION
**Pull request by phcorp**
| Q             | A
| ------------- | ---
| Branch?       | travis
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | --

### CODE REVIEW

File: rabbitmq.drush.inc [hunk 1]
```diff
@@ -80,7 +79,7 @@ function drush_rabbitmq_queue_info($queueName = NULL) {
   }
 
   /* @var \Drupal\Core\Queue\QueueFactory $queueFactory */
-  $queueFactory = (new Drupal())->service('queue');
+  $queueFactory = \Drupal::service('queue');
```
Comment (fgm) [inline rabbitmq.drush.inc]: drupalcs throws warnings on static method calls, hence the silly looking code creating a Drupal instance.

File: rabbitmq.drush.inc [hunk 2]
```diff
@@ -92,12 +91,12 @@ function drush_rabbitmq_queue_info($queueName = NULL) {
  */
 function drush_rabbitmq_worker($queueName) {
   /** @var \Drupal\rabbitmq\Consumer $consumer */
-  $consumer = (new Drupal())->service('rabbitmq.consumer');
+  $consumer = \Drupal::service('rabbitmq.consumer');
```
Comment (fgm) [inline rabbitmq.drush.inc]: idem

File: rabbitmq.drush.inc [hunk 3]
```diff
@@ -92,12 +91,12 @@ function drush_rabbitmq_queue_info($queueName = NULL) {
...
 
   // Service might be called from a non-Drush environment, so drush_get_option()
   // may not be available to it.
   $consumer->setOptionGetter(function(string $name) {
-    return intval(drush_get_option($name, Consumer::OPTIONS[$name]));
+    return (int) drush_get_option($name, Consumer::OPTIONS[$name]);
```
Comment (fgm) [inline rabbitmq.drush.inc]: Any advantage in using the cast over the function call ?

File: rabbitmq.drush.inc [hunk 4]
```diff
@@ -106,9 +105,15 @@ function drush_rabbitmq_worker($queueName) {
     $consumer->consume($queueName);
   }
   catch (InvalidWorkerException $e) {
-    return drush_set_error(dt("Worker for queue @name does not implement the worker interface.", $queueArgs));
+    return drush_set_error(dt(
+      'Worker for queue @name does not implement the worker interface.',
```
Comment (fgm) [inline rabbitmq.drush.inc]: Good idea to replace the double quotes by single quotes, but the line length is not limited outside arrays and comments, so no line wrapping is expected after the "dt("

File: rabbitmq.drush.inc [hunk 5]
```diff
@@ -106,9 +105,15 @@ function drush_rabbitmq_worker($queueName) {
...
+      $queueArgs
+    ));
   }
   catch (Exception $e) {
-    return drush_set_error(dt("Could not obtain channel for queue.", $queueArgs));
+    return drush_set_error(dt(
+      'Could not obtain channel for queue.',
```
Comment (fgm) [inline rabbitmq.drush.inc]: idem

File: src/ConnectionWrapper.php [hunk 6]
```diff
@@ -27,7 +27,7 @@
  *
  * This class is not tied with the Drupal Queue API.
  */
-class Connection {
+class ConnectionWrapper {
```
Comment (fgm) [inline src/ConnectionWrapper.php]: Any reason to rename that class ? There is no collision in the namespace, so what is the point ?

File: src/Consumer.php [hunk 7]
```diff
@@ -165,60 +166,26 @@ public function logStart() {
...
+    $maxIterations = $this->getOption(static::OPTION_MAX_ITERATIONS);
+    $memoryLimit = $this->getOption(static::OPTION_MEMORY_LIMIT);
+    $timeout = $this->getOption(static::OPTION_TIMEOUT);
+    $continueListening = TRUE;
     while ($continueListening) {
       try {
         $channel->basic_qos(NULL, 1, NULL);
-        $channel->basic_consume($queueName, '', FALSE, FALSE, FALSE, FALSE, $callback);
-
+        $channel->basic_consume(
```
Comment (fgm) [inline src/Consumer.php]: same as dt() calls above

File: src/Consumer.php [hunk 8]
```diff
@@ -241,18 +208,60 @@ public function consume(string $queueName): bool {
...
+  }
+
+  /**
+   * Get a channel from a given queue.
    *
    * Convert the various low-level known exceptions to module-level ones to make
    * it easier to catch cleanly.
    *
-   * @param \Drupal\rabbitmq\Queue\Queue $queue
+   * @param \Drupal\rabbitmq\Queue\queue $queue
```
Comment (fgm) [inline src/Consumer.php]: Uppercase needed for PSR4 compliance.

File: src/Consumer.php [hunk 9]
```diff
@@ -262,7 +271,7 @@ public function consume(string $queueName): bool {
    * @throws \Drupal\rabbitmq\Exception\OutOfRangeException
    * @throws \Drupal\rabbitmq\Exception\RuntimeException
    */
-  protected function getChannel(Queue $queue) {
+  protected function getChannel(queue $queue): AMQPChannel {
```
Comment (fgm) [inline src/Consumer.php]: idem

File: src/Consumer.php [hunk 10]
```diff
@@ -313,15 +321,16 @@ protected function getWorker(string $queueName): QueueWorkerInterface {
    * @return bool
    *   Did it ?
    */
-  protected function hitIterationsLimit(int $maxIterations, int $iteration) {
+  protected function hitIterationsLimit(int $maxIterations, int $iteration): bool {
```
Comment (fgm) [inline src/Consumer.php]: Much as I regret it, result type hints have been voted against in D8 coding standards, although static type parameter hints have been allowed. So...

File: src/Consumer.php [hunk 11]
```diff
@@ -313,15 +321,16 @@ protected function getWorker(string $queueName): QueueWorkerInterface {
...
     if ($maxIterations > 0 && $maxIterations <= $iteration) {
-      $this->logger->notice('RabbitMQ worker has reached max number of iterations: @count. Exiting.',
+      $this->logger->notice(
+        'RabbitMQ worker has reached max number of iterations: @count. Exiting.',
```
Comment (fgm) [inline src/Consumer.php]: same at "dt("

**Review by fgm — CHANGES_REQUESTED** [review]
Lots of interesting things, but some coding standards breaks. Could you split it in two, putting the simple reformatting things in one PR, then the main one refactoring the consume() method in another one to make it easier to reason about ?

File: rabbitmq.drush.inc (hunk 3 above)
Comment (phcorp) [reply rabbitmq.drush.inc]: According to PHP Inspections (EA Extended), type casting [can be up to 6 times faster](https://github.com/kalessil/phpinspectionsea/blob/master/src/test/resources/fixtures/api/type-casting-php4-functions.php).
https://3v4l.org/E7soS

File: src/ConnectionWrapper.php (hunk 6 above)
Comment (phcorp) [reply src/ConnectionWrapper.php]: The actual connection is stored inside the object instanciating this class, so the class was renamed for clarity. Elsewhere in the module, an instance of this class is stored in a `$connection_factory` variable whose name does not seem to suit with its actual purpose.

File: rabbitmq.drush.inc (hunk 3 above)
Comment (fgm) [reply rabbitmq.drush.inc]: Sold!

**Comment by phcorp** [comment]
I removed reformatting things outside the PR.

**Comment by fgm** [comment]
Rebased and merged manually into #14

**Pull request closed without merging**